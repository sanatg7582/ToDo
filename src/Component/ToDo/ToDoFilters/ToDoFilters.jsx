import React from "react";
import PropTypes from "prop-types";
import styles from "./ToDoFilters.module.scss";
import Button from "../../Shared/Button";
import Input from "../../Shared/Input";
/**
 * This function for return filter button
 * @param {func} handleOnClick
 * @param {number} filterActive
 * @param {func} setSearchtermfun
 * @param {func} searchterm
 * @returns Element
 */
const ToDoFilters = props => {
  const { handleOnClick, filterActive, setSearchtermfun, searchterm } = props;
  return (
    <div className={styles.filterContainer}>
      <div className={styles.buttonSection} >
        <Button
          buttonType="button"
          buttonName="All"
          btnClass={filterActive === 0 ? "success" : "primary"}
          handleOnClick={() => handleOnClick()}
          status={filterActive === 0 ? true : false}
          index={0}
        />
        &nbsp;&nbsp;&nbsp;&nbsp;
      <Button
          buttonType="button"
          buttonName="Active"
          btnClass={filterActive === 1 ? "success" : "primary"}
          handleOnClick={() => handleOnClick(1)}
          status={filterActive === 1 ? true : false}
          index={1}
        />
        &nbsp;&nbsp;&nbsp;&nbsp;
      <Button
          buttonType="button"
          buttonName="Is Completed"
          btnClass={filterActive === 2 ? "success" : "primary"}
          handleOnClick={() => handleOnClick(2)}
          status={filterActive === 2 ? true : false}
          index={2}
        />
      </div>
      <div className={styles.searchSection}>
        <Input InputType="text" InputName="searchterm" InputValue={searchterm} placeholder="search" handleChange={(e) => { setSearchtermfun(e.target.value) }} />
      </div>
    </div>
  );
};
// default props value
ToDoFilters.defaultProps = {
  handleOnClick: () => { },
  filterActive: 0
};
// prop type validation
ToDoFilters.propTypes = {
  handleOnClick: PropTypes.func.isRequired,
  filterActive: PropTypes.number
};
export default ToDoFilters;
