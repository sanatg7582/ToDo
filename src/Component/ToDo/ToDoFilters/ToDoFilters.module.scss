.filterContainer {
  padding: 10px 40px;
  display: flex;
  justify-content: space-between;
  .buttonSection{
    width: 75%;
  }
  .searchSection{
    width: 50%;
    margin-right: 68px;
  }
  input{
    width: 100%;
    margin-top: 10px;
  }
}
