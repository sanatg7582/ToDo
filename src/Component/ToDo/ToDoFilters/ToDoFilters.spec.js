import React from "react";
import { mount } from "enzyme";
import ToDoFilters from "./ToDoFilters";
import Input from "../../Shared/Input";
describe("Load <ToDoFilters />", () => {
  let wrapper = "";
  let setSearchtermfun = jest.fn();
  let handleOnClick = jest.fn();
  beforeAll(() => {
    const props = {
      handleOnClick: handleOnClick,
      filterActive: 0,
      setSearchtermfun: setSearchtermfun,
      searchterm: ""
    };
    wrapper = mount(<ToDoFilters {...props} />);
  });
  it("renders without crashing", () => {
    expect(wrapper).toBeTruthy();
  });
  it("checking TodoFilter length", () => {
    expect(wrapper).toHaveLength(1);
    expect(wrapper.find("button")).toHaveLength(3);
  });
  it("Checking button props", () => {
    expect(
      wrapper
        .find("button")
        .first()
        .props().className
    ).toEqual("btn success");
    wrapper.setProps({ filterActive: 1 });
    expect(
      wrapper
        .find("button")
        .first()
        .props().className
    ).toEqual("btn primary");
    wrapper.setProps({ filterActive: 2 });
    expect(
      wrapper
        .find("button")
        .first()
        .props().className
    ).toEqual("btn primary");
    expect(
      wrapper
        .find("button")
        .at(1)
        .props().className
    ).toEqual("btn primary");
    expect(
      wrapper
        .find("button")
        .at(2)
        .props().className
    ).toEqual("btn success");
  });

  it("Checking search input class and value", () => {
    expect(wrapper.find(Input).find("input").props().value).toEqual("");
    expect(wrapper.find(Input).find("input").props().className).toEqual("formInput");
  });

  it("Checking Search onchange", () => {
    wrapper.find(Input).find("input").simulate("change");
    expect(setSearchtermfun).toHaveBeenCalled();
  });

  it("SnapShot test", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
