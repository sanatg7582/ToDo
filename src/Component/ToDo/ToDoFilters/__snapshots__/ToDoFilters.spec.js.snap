// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Load <ToDoFilters /> SnapShot test 1`] = `ReactWrapper {}`;
